tests/foreman/api/test_hostcollection.py

```diff
@@ -23,438 +23,464 @@
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
-from robottelo.test import APITestCase
 from robottelo.vm import VirtualMachine
 
 
-class HostCollectionTestCase(APITestCase):
-    """Tests for host collections."""
-
-    @classmethod
-    def setUpClass(cls):
-        """Create hosts that can be shared by tests."""
-        super().setUpClass()
-        cls.org = entities.Organization().create()
-        cls.hosts = [entities.Host(organization=cls.org).create() for _ in range(2)]
-        cls.lce = entities.LifecycleEnvironment(organization=cls.org).create()
-        cls.content_view = entities.ContentView(organization=cls.org).create()
-        cls.content_view.publish()
-        cls.content_view = cls.content_view.read()
-        promote(cls.content_view.version[0], environment_id=cls.lce.id)
-        cls.activation_key = entities.ActivationKey(
-            content_view=cls.content_view, environment=cls.lce, organization=cls.org
-        ).create()
+@pytest.fixture(scope='module')
+def fake_hosts(module_org):
+    """Create content hosts that can be shared by tests."""
+    hosts = [entities.Host(organization=module_org).create() for _ in range(2)]
+    return hosts
+
+
+@pytest.fixture(scope='module')
+def content_view(module_org, module_lce, module_published_cv):
+    content_view = module_published_cv
+    content_view = content_view.read()
+    promote(content_view.version[0], environment_id=module_lce.id)
+    return content_view
+
+
+@pytest.fixture(scope='module')
+def activation_key(module_org, module_published_cv, module_lce):
+    activation_key = entities.ActivationKey(
+        content_view=module_published_cv, environment=module_lce, organization=module_org
+    ).create()
+    return activation_key
+
+
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+@pytest.mark.tier1
+def test_positive_create_with_name(module_org, name):
+    """Create host collections with different names.
+
+    :id: 8f2b9223-f5be-4cb1-8316-01ea747cae14
+
+    :parametrized: yes
+
+    :expectedresults: The host collection was successfully created and has
+        appropriate name.
+
+    :CaseImportance: Critical
+    """
+    host_collection = entities.HostCollection(name=name, organization=module_org).create()
+    assert host_collection.name == name
+
+
+@pytest.mark.tier1
+def test_positive_list(module_org):
+    """Create new host collection and then retrieve list of all existing
+    host collections
+
+    :id: 6ae32df2-b917-4830-8709-15fb272b76c1
+
+    :BZ: 1331875
+
+    :expectedresults: Returned list of host collections for the system
+        contains at least one collection
+
+    :CaseImportance: Critical
+    """
+    entities.HostCollection(organization=module_org).create()
+    hc_list = entities.HostCollection().search()
+    assert len(hc_list) >= 1
+
+
+@pytest.mark.tier1
+def test_positive_list_for_organization(module_org):
```

Comment:
I think module_org is not required here.

Reply:
OK